reverserepl/activity/prepare_dataflow_writer.go

```diff
@@ -0,0 +1,121 @@
+package activity
+
+import (
+	"context"
+	"fmt"
+
+	dataflowaccessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/dataflow"
+	dataflowutils "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/utils/dataflow"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	resource "github.com/GoogleCloudPlatform/spanner-migration-tool/reverserepl/resource"
+)
+
+type PrepareDataflowWriterInput struct {
+	SmtJobId               string
+	SourceShardsFilePath   string
+	SessionFilePath        string
+	SourceType             string
+	SourceDbTimezoneOffset string
+	TimerInterval          int
+	StartTimestamp         string
+	WindowDuration         string
+	GCSInputDirectoryPath  string
+	SpannerProjectId       string
+	MetadataInstance       string
+	MetadataDatabase       string
+	MetadataTableSuffix    string
+	TuningCfg              string
+	SpannerLocation        string
+}
+
+type PrepareDataflowWriterOutput struct {
+	JobId string
+}
+
+type PrepareDataflowWriter struct {
+	Input  *PrepareDataflowWriterInput
+	Output *PrepareDataflowWriterOutput
+}
+
+// Launches the writer dataflow job.
+func (p *PrepareDataflowWriter) Transaction(ctx context.Context) error {
+	input := p.Input
+	writerTuningCfg, err := dataflowutils.UnmarshalDataflowTuningConfig(ctx, input.TuningCfg)
+	if err != nil {
+		return fmt.Errorf("error reading writer tuning config %s: %v", input.TuningCfg, err)
+	}
+	logger.Log.Debug(fmt.Sprintf("writerTuningCfg: %+v", writerTuningCfg))
+	validateUpdateWriterTuningCfg(&writerTuningCfg, input.SpannerProjectId, input.SpannerLocation, input.SmtJobId)
+	logger.Log.Debug(fmt.Sprintf("Updated writerTuningCfg: %+v", writerTuningCfg))
+	params := map[string]string{
+		"sourceShardsFilePath":   input.SourceShardsFilePath,
+		"sessionFilePath":        input.SessionFilePath,
+		"sourceType":             input.SourceType,
+		"sourceDbTimezoneOffset": input.SourceDbTimezoneOffset,
+		"timerInterval":          fmt.Sprintf("%v", input.TimerInterval),
+		"windowDuration":         input.WindowDuration,
+		"GCSInputDirectoryPath":  input.GCSInputDirectoryPath,
+		"metadataTableSuffix":    input.MetadataTableSuffix,
+		"spannerProjectId":       input.SpannerProjectId,
+		"metadataInstance":       input.MetadataInstance,
+		"metadataDatabase":       input.MetadataDatabase,
+		"startTimestamp":         input.StartTimestamp,
+		"runIdentifier":          input.SmtJobId,
+		"runMode":                constants.RR_WRITER_REGULAR_MODE,
+	}
+	dfLaunchReq, err := dataflowutils.GetDataflowLaunchRequest(params, writerTuningCfg)
+	if err != nil {
+		return err
+	}
+	dfJobId, err := resource.CreateDataflowSMTResource(ctx, input.SmtJobId, dfLaunchReq)
+	if err != nil {
+		return err
+	}
+	logger.Log.Info(fmt.Sprintf("Launched writer job with id: %s", dfJobId))
```

Comment:
It will be good to print the gcloud command as well, it will be easier for the user than querying it from the metadata tables